Applications/Spire/Include/Spire/Ui/TitleBar.hpp

```diff
@@ -24,19 +23,19 @@ namespace Spire {
        */
       void set_icon(const QImage& icon);
 
-      /** Returns the title text QLabel. */
-      QLabel* get_title_label() const;
+      /** Returns the title text label. */
+      TextBox* get_title_label() const;
```

Comment:
If this can't be null, then return a reference. Consider whether the state of the ```TextBox``` is part of the substate of the ```TitleBar``` as a whole. If it is part of the substate then you must return a ```const TextBox&```, if it isn't then you can keep it as a ```TextBox&```.

Reply:
Can you explain what you mean by substate? I would look at this from an API perspective, where I would return a const reference because nothing else should need to modify the TitleBar.

Reply:
The ```TitleBar``` class has a set of state that makes up a title bar. Is the state of the ```TextBox``` returned by ```get_title_label``` a subset of the state of the ```TitleBar```? For example consider a ```shared_ptr<int>``` which is a shared pointer to an integer. The state of the integer is not a part of the state of the ```shared_ptr<int>```, and hence ```shared_ptr<int>::get() const``` returns an ```int*``` instead of an ```const int*```. Changing the ```int``` does not change the state of the ```shared_ptr<int>```, the states are non-overlapping. If the state of the ```TextBox``` is non-overlapping with the state of the ```TitleBar```, then return a ```TextBox&```, if there is overlap then return a ```const TextBox&```.